.github/workflows/doc-build.yml

```diff
@@ -0,0 +1,61 @@
+name: Quarkus Documentation CI
+
+on:
+  push:
+    paths:
+      - 'docs/src/main/asciidoc/**'
+  pull_request:
+    paths:
+      - 'docs/src/main/asciidoc/**'
+
+
+jobs:
+  build-doc:
+    name: "Documentation Build"
+    runs-on: ubuntu-latest
+    # Skip draft PRs and those with WIP in the subject, rerun as soon as its removed
+    if: "github.event_name != 'pull_request' || ( \
+             github.event.pull_request.draft == false && \
+             github.event.pull_request.state != 'closed' && \
+             contains(github.event.pull_request.title, 'wip ') == false && \
+             contains(github.event.pull_request.title, '[wip]') == false && \
+             (
+               github.event.action != 'edited' || \
+               contains(github.event.changes.title.from, 'wip ') || \
+               contains(github.event.changes.title.from, '[wip]') \
+             ) \
+           )"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: n1hility/cancel-previous-runs@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up JDK 11
+        # Uses sha for added security since tags can be updated
+        uses: joschi/setup-jdk@b9cc6eabf7e7e3889766b5cee486f874c9e1bd2d
+        with:
+          java-version: 11
+      - name: Compute cache restore key
+        # Always recompute on a push so that the maven repo doesnt grow indefinitely with old versions
+        run: |
+          if ${{ github.event_name == 'pull_request' }}; then echo "::set-env name=COMPUTED_RESTORE_KEY::q2maven-"; fi
+      - name: Cache Maven Repository
+        id: cache-maven
+        uses: n1hility/cache@v2
+        with:
+          path: ~/.m2/repository
+          # Improves the reusability of the cache to limit key changes
+          key: q2maven-${{ hashFiles('bom/runtime/pom.xml') }}
+          restore-keys: ${{ env.COMPUTED_RESTORE_KEY }}
+          restore-only: ${{ github.event_name == 'pull_request' }}
+      - name: Build
+        run: |
+          mvn -e -B --settings .github/doc-mvn-settings.xml clean package -pl docs
```

Comment:
Why a specific mvn settings?

Reply:
To include the Sonatype repo to download Quarkus snapshots instead of building the full platform

Reply:
Why would you need the snapshots to build the doc?

Reply:
if it's about the `AllConfigGenerator`, I think I would just run the Asciidoctor plugin for this step.

Reply:
It also needs some quarkus plugin. I'll have a look, if I direclty use the asciidoctor plugin maybe it's not needed

Reply:
The doc project uses the extension processor so we need the snapshot repo or the pom.xml will not be usable.
```
Failed to collect dependencies at io.quarkus:quarkus-extension-processor:jar:999-SNAPSHOT
```

Reply:
As there is two Quarkus depedencies inside the doc `pom.xml` there is no choice. Even if I launch the asciidoc plugin manually there will be some dependency resolution so I will need a repository to fetch the snapshots.

Reply:
I just tested and using directly the plugin skip the dependencies downloading so it works without a dedicated Maven settings.
And the Job is now executed in less than a minute, see https://github.com/quarkusio/quarkus/pull/10428/checks?check_run_id=830891662.

If it's OK, I'll remove the fake doc upgrade and finalize this PR.